modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -343,6 +390,61 @@ public SCIMResponse listWithGET(UserManager userManager, String filter, int star
         }
     }
 
+    /**
+     * Method to process a list and return a SCIM response.
+     *
+     * @param tempList          Filtered user list
+     * @param encoder           Json encoder
+     * @param attributes        Required attributes
+     * @param excludeAttributes Exclude attributes
+     * @param startIndex        Starting index
+     * @return SCIM response
+     * @throws NotFoundException
+     * @throws CharonException
+     * @throws BadRequestException
+     */
+    private SCIMResponse processGroupList(List<Object> tempList, JSONEncoder encoder, String attributes,
+            String excludeAttributes, int startIndex) throws NotFoundException, CharonException, BadRequestException {
+
+        int totalResults = 0;
+        List<Object> returnedGroups;
+        if (tempList == null) {
+            tempList = Collections.emptyList();
+        }
+        try {
+            totalResults = (int) tempList.get(0);
+            tempList.remove(0);
+        } catch (IndexOutOfBoundsException e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Group result list is empty.");
+            }
+            totalResults = tempList.size();
+        } catch (ClassCastException ex) {
+            if (logger.isDebugEnabled()) {
```

Comment:
we don't want to throw ClassCastException if we do the cast check before casting? instanceOf

Reply:
fixed: https://github.com/wso2/charon/pull/263/commits/61f4e744f53695f16e5ad6c9496f369ff77101f8